boa_engine/src/object/jsobject.rs

```diff
@@ -1102,10 +1101,29 @@ impl Debug for JsObject {
         // Instead, we check if the object has appeared before in the entire graph. This means that objects will appear
         // at most once, hopefully making things a bit clearer.
         if !limiter.visited && !limiter.live {
-            f.debug_struct("JsObject")
-                .field("vtable", &ptr)
-                .field("object", &self.inner.object)
-                .finish()
+            let ptr: *const _ = self.as_ref();
+            let obj = self.borrow();
+            let kind = obj.kind();
+            let name_prop = self
+                .borrow()
+                .properties()
+                .get(&PropertyKey::String(JsString::from("name")));
+
+            if obj.is_function() {
```

Comment:
```suggestion
            if obj.is_function() {            
                let name_prop = obj
                    .properties()
                    .get(&PropertyKey::String(JsString::from("name")));
```

Since we only use it for functions only, we can move it, also we already borrowed it so we can use the `obj`